cli/command/formatter/namespaces.go

```diff
@@ -0,0 +1,266 @@
+	h := r.Replace(names[len(names)-1])
+
+	return h
+}
+
+type namespacesContext struct {
+	HeaderContext
+	trunc     bool
+	c         types.Container
+	apiClient client.APIClient
+
+	procPath string
+
+	cJSON *types.ContainerJSON
+}
+
+func newNamespacesContext(client client.APIClient, t bool, c types.Container) *namespacesContext {
+	return &namespacesContext{
+		apiClient: client,
+		trunc:     t,
+		procPath:  "/proc", /* default proc path */
+		c:         c,
+	}
+}
+
+func (n *namespacesContext) FullHeader() interface{} {
+	return n.header
+}
+
+func (n *namespacesContext) MarshalJSON() ([]byte, error) {
+	return marshalJSON(n)
+}
+
+func (n *namespacesContext) ID() string {
+	if n.trunc {
+		return stringid.TruncateID(n.c.ID)
+	}
+
+	return n.c.ID
+}
+
+func (n *namespacesContext) containerJSON() (types.ContainerJSON, error) {
+	if n.cJSON != nil {
+		return *n.cJSON, nil
+	}
+
+	c, err := n.apiClient.ContainerInspect(context.Background(), n.c.ID)
+	if err != nil {
+		return c, err
+	}
+
+	n.cJSON = &c
+
+	return c, nil
+}
+
+func (n *namespacesContext) PID() string {
+	c, err := n.containerJSON()
+	if err != nil {
+		return err.Error()
+	}
+
+	pid, err := n.stat(c.State.Pid, "pid")
+	if err != nil {
+		return err.Error()
+	}
+
+	return pid
+}
+
+func (n *namespacesContext) PATH() string {
+	c, err := n.containerJSON()
+	if err != nil {
+		return err.Error()
+	}
+
+	return c.Path
+}
+
+func (n *namespacesContext) CGROUP() string {
+	c, err := n.containerJSON()
+	if err != nil {
+		return err.Error()
+	}
+
+	cgroup, err := n.stat(c.State.Pid, "cgroup")
+	if err != nil {
+		return err.Error()
+	}
+
+	return cgroup
+}
+
+func (n *namespacesContext) IPC() string {
+	c, err := n.containerJSON()
+	if err != nil {
+		return err.Error()
+	}
+
+	ipc, err := n.stat(c.State.Pid, "ipc")
+	if err != nil {
+		return err.Error()
+	}
+
+	return ipc
+}
+
+func (n *namespacesContext) NAME() string {
+	return n.c.Names[0]
+}
+
+func (n *namespacesContext) MNT() string {
+	c, err := n.containerJSON()
+	if err != nil {
+		return err.Error()
+	}
+
+	mnt, err := n.stat(c.State.Pid, "mnt")
+	if err != nil {
+		return err.Error()
+	}
+
+	return mnt
+}
+
+func (n *namespacesContext) NET() string {
+	c, err := n.containerJSON()
+	if err != nil {
+		return err.Error()
+	}
+
+	mnt, err := n.stat(c.State.Pid, "net")
+	if err != nil {
+		return err.Error()
+	}
+
+	return mnt
+}
+
+func (n *namespacesContext) USER() string {
+	c, err := n.containerJSON()
+	if err != nil {
+		return err.Error()
+	}
+
+	mnt, err := n.stat(c.State.Pid, "user")
+	if err != nil {
+		return err.Error()
+	}
+
+	return mnt
+}
+
+func (n *namespacesContext) UTS() string {
+	c, err := n.containerJSON()
+	if err != nil {
+		return err.Error()
+	}
+
+	mnt, err := n.stat(c.State.Pid, "uts")
+	if err != nil {
+		return err.Error()
+	}
+
+	return mnt
+}
+
+func (n *namespacesContext) stat(pid int, ns string) (string, error) {
+	o, err := os.Readlink(n.procPath + "/" + strconv.Itoa(pid) + "/ns/" + ns)
```

Comment:
This will not work because it assumes the cli is running on the same host as the API, which is not always the case.

Reply:
Do you recommend to gather this info in API side ?

I assumed this feature is only for Linux platforms.
https://github.com/moby/moby/issues/32501#issuecomment-293359638:
_On non-Linux, it would report "This option only supported on Linux."_

Reply:
It's not the platform that's the problem, but it's the assumption that the CLI is running on the same host (what if the host is remote?)